### PR TITLE
Homework2: Alexey Masharov

### PR DESCRIPTION
**Pull request by JIekceu**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle).
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks.
- [ ] my PR made from the fork feature branch to appropriate branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore.

### CODE REVIEW

File: src/Homework2/task1.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+1. Напишите программу, которая считает общую цену.
+Вводится M рублей и N копеек цена, а также количество S товара
+Посчитайте общую цену в рублях и копейках за L товаров.
+Пример:
+Input: Цена одной вещи 3 рубля 20 копеек, посчитать 3 предмета.
+Output: Общая цена 9 рублей 60 копеек
+Задачу поместите в файл task1.py в папке src/homework2.
+'''
+while True:
+    m = int(input("Цена товара, руб.: "))
```
Comment (belovmd) [inline src/Homework2/task1.py:11]: too complex

File: src/Homework2/task1.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿'''
+1. Напишите программу, которая считает общую цену.
+Вводится M рублей и N копеек цена, а также количество S товара
+Посчитайте общую цену в рублях и копейках за L товаров.
+Пример:
+Input: Цена одной вещи 3 рубля 20 копеек, посчитать 3 предмета.
+Output: Общая цена 9 рублей 60 копеек
+Задачу поместите в файл task1.py в папке src/homework2.
+'''
+while True:
```
Comment (belovmd) [inline src/Homework2/task1.py:10]: why do we need while here?

File: src/Homework2/task2.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+Подсказки:
+-	my_string.split([chars]) возвращает список строк.
+-	len(list) - количество элементов в списке
+Задачу поместите в файл task2.py в папке src/homework2.
+
+'''
+import re
+while True:
+    sentence = str(input("Введите строку: "))
+    sentence_cleared = re.sub('[^ A-Za-z0-9]', '', sentence)
```
Comment (belovmd) [inline src/Homework2/task2.py:13]: what about $ symbol and many others?

File: src/Homework2/task4.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+stroch = int()
+str_ = str(input("Введите строку: "))
+for i in str_:
+    if 'A' <= i <= 'Z':
+        propis += 1
+    if 'a' <= i <= 'z':
+        stroch += 1
+print(f'{propis} прописных букв, {stroch} строчных букв')
+dict_ = {"propis": 0, "stroch": 0}
+str_ = str(input("Введите строку: "))
```
Comment (belovmd) [inline src/Homework2/task4.py:8]: please leave only one solution and remove redundant code

File: src/Homework2/task5.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+            if i == n:
+                print(a)
+        else:
+            b = b + a
+            if i == n:
+                print(b)
+'''
+while True:
+    n = int(input("Введите n: "))
+    spisok = []
```
Comment (belovmd) [inline src/Homework2/task5.py]: redundant comments

File: src/Homework2/task5.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+Задачу поместите в файл task5.py в папке src/homework2.
+'''
+while True:
+    n = int(input("Введите n: "))
+    a = int(1)
+    b = int(1)
+    if n == 1 or n == 2:
+        print(1)
+    for i in range(3, n + 1):
+        if i % 2 > 0:
```
Comment (belovmd) [inline src/Homework2/task5.py:13]: too complex

File: src/Homework2/task6.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import copy
```
Comment (belovmd) [inline src/Homework2/task6.py]: missed task description

File: src/Homework2/task6.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+while True:
+    chislo = int(input('Введите число: '))
+    lst_ = []
```
Comment (belovmd) [inline src/Homework2/task6.py]: lists are forbidden

File: src/Homework2/task7.py [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+Требуется: проверить, действительно ли это стороны треугольника.
+Если стороны определяют треугольник, найти его площадь. Если
+нет, вывести сообщение о неверных данных.
+'''
+s = float()
+while True:
+    a = float(input("Введите длину первой стороны: "))
+    b = float(input("Введите длину второй стороны: "))
+    c = float(input("Введите длину третьей стороны: "))
+    if a + b > c or a + c > b or b + c > a:  # тест на треугольник
```
Comment (belovmd) [inline src/Homework2/task7.py:12]: redundant comment

**Review by belovmd — CHANGES_REQUESTED** [review]
missed task 8

File: src/Homework2/task1.py (hunk 2 above)
Comment (JIekceu) [reply src/Homework2/task1.py:10]: To check working program with some different input data without restart program

File: src/Homework2/task2.py (hunk 3 above)
Comment (JIekceu) [reply src/Homework2/task2.py:13]: All characters except A-Za-z0-9 are removed 
'^' - except

File: src/Homework2/task6.py (hunk 7 above)
Comment (JIekceu) [reply src/Homework2/task6.py]: fixed

File: src/Homework2/task6.py (hunk 8 above)
Comment (JIekceu) [reply src/Homework2/task6.py]: fixed

File: src/Homework2/task7.py (hunk 9 above)
Comment (JIekceu) [reply src/Homework2/task7.py:12]: fixed

File: src/Homework2/task4.py (hunk 4 above)
Comment (JIekceu) [reply src/Homework2/task4.py:8]: fixed

**Comment by belovmd** [comment]
9

**Pull request closed without merging**